### PR TITLE
feat: stub for create dependent features

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

* create and read schema for feature dependencies
* endpoint for upserting (create or update) a single feature dependency
* added a flag for featureDependencies
* API for the service (no impl yet)
* a test showing all the pieces fit

### CODE REVIEW

File: src/lib/openapi/spec/dependent-feature-schema.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { FromSchema } from 'json-schema-to-ts';
+import { createDependentFeatureSchema } from './create-dependent-feature-schema';
+
+export const dependentFeatureSchema = {
+    $id: '#/components/schemas/dependentFeatureSchema',
+    type: 'object',
+    description: 'Feature dependency on a parent feature in read model',
+    required: ['feature'],
+    additionalProperties: false,
```
Comment (kwasniew) [inline src/lib/openapi/spec/dependent-feature-schema.ts:9]: same as create schema but without additional properties

File: src/lib/routes/admin-api/project/project-features.ts [hunk 2]
```diff
@@ -297,6 +299,29 @@ export default class ProjectFeaturesController extends Controller {
             ],
         });
 
+        this.route({
+            method: 'post',
+            path: PATH_DEPENDENCIES,
+            handler: this.addFeatureDependency,
+            permission: CREATE_FEATURE,
```
Comment (kwasniew) [inline src/lib/routes/admin-api/project/project-features.ts:306]: decided not to introduce new permissions for now and use existing create feature permission

File: src/lib/services/feature-toggle-service.ts [hunk 3]
```diff
@@ -2201,6 +2211,27 @@ class FeatureToggleService {
...
+                      child: feature,
+                      enabled,
+                  }
+                : {
+                      parent: parentFeature,
+                      child: feature,
+                      enabled: true,
+                      variants,
+                  };
+        console.log(featureDependency);
```
Comment (kwasniew) [inline src/lib/services/feature-toggle-service.ts:2233]: deliberate console log for now

File: src/test/e2e/api/admin/project/features.dependencies.e2e.test.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        .expect(expectedCode);
+};
+
+test('should add feature dependency', async () => {
+    const parent = uuidv4();
+    const child = uuidv4();
+    await app.createFeature(parent);
+    await app.createFeature(child);
+
+    await addFeatureDependency(parent, {
```
Comment (kwasniew) [inline src/test/e2e/api/admin/project/features.dependencies.e2e.test.ts:53]: checking if we don't return any errors

**Review by FredrikOseberg — APPROVED** [review]
LGTM

**Pull request merged**